handler/src/main/java/io/netty5/handler/ssl/SslHandshakeCompletionEvent.java

```diff
@@ -15,24 +15,53 @@
  */
 package io.netty5.handler.ssl;
 
+import javax.net.ssl.SSLSession;
+import java.util.Objects;
+
 /**
  * Event that is fired once the SSL handshake is complete, which may be because it was successful or there
  * was an error.
  */
-public final class SslHandshakeCompletionEvent extends SslCompletionEvent {
-
-    public static final SslHandshakeCompletionEvent SUCCESS = new SslHandshakeCompletionEvent();
+public final class SslHandshakeCompletionEvent
+        extends SslCompletionEvent<SslHandshakeCompletionEvent.SslHandshakeData> {
```

Comment:
I was wondering if this should just be `extends SslCompletionEvent<SSLEngine>`. The downside would be that it would be easy for people to mess with the `SSLEngine` itself while generally speaking they should not. The upside would be that the engine also has methods like `getApplicationProtocol()`. So maybe `SSLEngine` would be better ?

Reply:
ok so lets keep it like it is for now... I guess that is also better as `SSLEngine` might be reference-counted